### PR TITLE
CICD: Add create-release-pr workflow

### PR DESCRIPTION
**Pull request by algobarb**
## Summary
Add create-release-pr workflow to create release PRs using workflow dispatch.

## Testing
In separate repo:
- Made sure updated Changelog.md looks right
- Made sure README gets updated
- tags are added with a hyphen

### CODE REVIEW

File: scripts/bump_version.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This script bumps up the version in setup.py for new releases.
```
Comment (excalq) [inline scripts/bump_version.py:1]:
```suggestion
# This script bumps up the version in `pom.xml` for new releases.
```

File: scripts/bump_version.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Usage: python bump_version.py {new_version} (--read_me <path-to-README.md> --pom_xml <path-to-pom-xml>)
+
+import argparse
+import re
+
+def bump_version(new_version, file_path):
+    with open(file_path, "r") as file:
+        read_me = file.read()
```
Comment (excalq) [inline scripts/bump_version.py:9]: nit: It's only 'README' for one invocation. Generalize variable name.
```suggestion
        content = file.read()
```

**Review by excalq — APPROVED** [review]
Only the comment update and var-name nit, both non-blocking.

File: scripts/bump_version.py (hunk 2 above)
Comment (algobarb) [reply scripts/bump_version.py]: Yes thank you

Comment (algobarb) [reply scripts/bump_version.py]: Updated.

**Pull request merged**